draft-ietf-quic-transport.md

```diff
@@ -1378,28 +1378,38 @@ A server MUST include the original_connection_id transport parameter
 ({{transport-parameter-definitions}}) if it sent a Retry packet to enable
 validation of the Retry, as described in {{packet-retry}}.
 
-
 ### Values of Transport Parameters for 0-RTT {#zerortt-parameters}
 
-A client that attempts to send 0-RTT data MUST remember the transport parameters
-used by the server.  The transport parameters that the server advertises during
-connection establishment apply to all connections that are resumed using the
-keying material established during that handshake.  Remembered transport
+The transport parameters that the server advertises during connection
+establishment generally apply to all connections that are resumed using
+the keying material established during that handshake.  Remembered transport
 parameters apply to the new connection until the handshake completes and new
 transport parameters from the server can be provided.
 
-A server can remember the transport parameters that it advertised, or store an
-integrity-protected copy of the values in the ticket and recover the information
-when accepting 0-RTT data.  A server uses the transport parameters in
-determining whether to accept 0-RTT data.
-
-A server MAY accept 0-RTT and subsequently provide different values for
-transport parameters for use in the new connection.  If 0-RTT data is accepted
-by the server, the server MUST NOT reduce any limits or alter any values that
-might be violated by the client with its 0-RTT data.  In particular, a server
-that accepts 0-RTT data MUST NOT set values for the following parameters
-({{transport-parameter-definitions}}) that are smaller
-than the remembered value of those parameters.
+The value of the server's previous original_connection_id, preferred_address,
+stateless_reset_token, and ack_delay_exponent MUST NOT be used when
+establishing a new connection; rather, the client should wait to observe the
+server's new values in the handshake.
+
+The client MAY store the server's original max_ack_delay and max_packet_size
```

Comment:
max_packet_size is important.  That is the one in addition to the flow control stuff that needs to be respected.

Reply:
I figure the client always has the option of using 1200B packets until the handshake's done.

Reply:
The client can do that for Handshake and Initial, but not 1-RTT.  0-RTT should follow the same rules though.

Reply:
I don't understand your comment. The safe thing to do is to send 0-RTT at 1280 until a new probe can be sent. I can also understand the opposite choice, which is why this is a MAY.

Reply:
The point of max_packet_size is to limit the size of packets.  I realize that if you are making a new connection, the limit applies only to 1-RTT packets because Initial and Handshake packets are sent without knowing what the limits are (the server could comply with a request to send smaller Handshake packets, and maybe the client could also, but it's not something we can guarantee).  However, when sending 0-RTT, you know what the limit was, so why not respect it?

Reply:
OK, I'll make it a MUST